packages/popover/src/vaadin-popover.js

```diff
@@ -577,16 +583,84 @@ class Popover extends PopoverPositionMixin(
       event.stopPropagation();
       this._openedStateController.close(true);
     }
+
+    // Include popover content in the Tab order after the target.
+    if (event.key === 'Tab') {
+      if (event.shiftKey) {
+        this.__onGlobalShiftTab(event);
+      } else {
+        this.__onGlobalTab(event);
+      }
+    }
   }
 
   /** @private */
-  __onTargetKeydown(event) {
-    // Prevent restoring focus after target blur on Tab key
-    if (event.key === 'Tab' && this.__shouldRestoreFocus) {
-      this.__shouldRestoreFocus = false;
+  __onGlobalTab(event) {
+    const overlayPart = this._overlayElement.$.overlay;
+
+    // Move focus to the popover content on target element Tab
+    if (this.target && isElementFocused(this.target)) {
+      event.preventDefault();
+      overlayPart.focus();
+      return;
+    }
+
+    // Move focus to the next element after target on content Tab
+    const lastFocusable = this.__getLastFocusable(overlayPart);
+    if (lastFocusable && isElementFocused(lastFocusable)) {
+      const focusable = this.__getNextBodyFocusable(this.target);
+      if (focusable && focusable !== overlayPart) {
+        event.preventDefault();
+        focusable.focus();
+        return;
+      }
```

Comment:
Could this be changed to the following to make it rely on native tabbing order?
```js
this.target.focus();
return;
```

Reply:
Good point, changed.

Reply:
I'm kind of having second thoughts about this suggestion. Shift-tabbing back to the overlay needs to rely on the programmatic identification of the next tabbable. If at the same time tabbing from the overlay to the next tabbable relies on native tab order, in the worst case, we might end up with a scenario where you tab away from the overlay, and then try to shift-tab back to it, but the focus ends up somewhere else. If you see this as an issue, please revert the related change.